### ISSUE TITLE
Update React TransitionsGroup To solve "componentWillMount has been renamed"

### ISSUE THREAD

**Issue by el-j** [tjrexer/react-notifications#12]
React TransitionGroup is already updated. Here is a really old dep in use -
i checked the guid and changed the Notifications.js the use the TransistionGroup
would be nice to get it updated. 

Thanks

![image](https://user-images.githubusercontent.com/2795534/77705145-34977800-6fbf-11ea-8e3b-b0e946485a39.png)

Mirgation Guide:
https://github.com/reactjs/react-transition-group/blob/master/Migration.md

src/Notifications.js
```
import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import classnames from 'classnames';
import Notification from './Notification';

class Notifications extends React.Component {
  static propTypes = {
    notifications: PropTypes.array.isRequired,
    onRequestHide: PropTypes.func,
    enterTimeout: PropTypes.number,
    leaveTimeout: PropTypes.number
  };

  static defaultProps = {
    notifications: [],
    onRequestHide: () => {
    },
    enterTimeout: 400,
    leaveTimeout: 400
  };

  handleRequestHide = notification => () => {
    const { onRequestHide } = this.props;
    if (onRequestHide) {
      onRequestHide(notification);
    }
  };

  render() {
    const { notifications, enterTimeout, leaveTimeout } = this.props;
    const className = classnames('notification-container', {
      'notification-container-empty': notifications.length === 0
    });

    const items = notifications.map((notification,i) => {
        const key = notification.id || new Date().getTime();
        return (
          <CSSTransition
            key={i}
            classNames="notification"
            timeout={{ enter: enterTimeout, exit: leaveTimeout }}
          >
          <Notification
            key={key}
            type={notification.type}
            title={notification.title}
            message={notification.message}
            timeOut={notification.timeOut}
            onClick={notification.onClick}
            onRequestHide={this.handleRequestHide(notification)}
          />
          </CSSTransition>
        );
      })
    return (
      <div className={className}>

        <TransitionGroup >
          {items}
        </TransitionGroup>
      </div>
    );
  }
}

export default Notifications;
```

**Comment by tjrexer** [comment]
@el-j I didn't realize this is a duplicate of #11.  This has been resolved in [fa4f4c7](https://github.com/tjrexer/react-notifications/commit/fa4f4c71a2678c085d2a663ab092bb3dbe00c30e).  However, I did like the additional changes you made for updating the other requirements.  If you can resolve the gulp issues, please do so and submit a new PR for next as I need to update all of the requirements eventually.

**Comment by fjmoralesp** [comment]
Hi!

is this solved?, react-notifications updated to avoid deprecation message, however I'm using `"react-notifications": "^1.6.0"` and the warning message still appears.

and it looks like there is another issue related with react running in strict mode:

<img width="579" alt="Captura de Pantalla 2020-04-11 a la(s) 8 49 21 a  m" src="https://user-images.githubusercontent.com/8378043/79045575-76383d80-7bd1-11ea-8dfa-f4055ce960a9.png">

**Comment by tjrexer** [comment]
Resolved in 1.7.0
